### PR TITLE
Fix known issue and bump API version

### PR DESCRIPTION
**Pull request by TracyYaoYao**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Cdn/custom/Remove-AzCdnProfile.ps1 [hunk 1]
```diff
@@ -155,6 +155,10 @@ function Remove-AzCdnProfile {
     )
     
     process {
+        $hasAsJob = $PSBoundParameters.Remove('AsJob')
```
Comment (msJinLei) [inline src/Cdn/custom/Remove-AzCdnProfile.ps1:158]: It is not a good idea to remove and add back parameters.
I propose to copy the $PSBoundParameters to a new variable for Get-AzCdnProfile

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Cdn/custom/Remove-AzCdnProfile.ps1 (hunk 1 above)
Comment (TracyYaoYao) [reply src/Cdn/custom/Remove-AzCdnProfile.ps1:158]: Copy. Thanks for you advice. Will fix.

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Cdn/custom/Start-AzFrontDoorCdnProfilePrepareMigration.ps1 [hunk 2]
```diff
@@ -297,11 +297,11 @@ function Start-AzFrontDoorCdnProfilePrepareMigration {
         
         # Deal with MSI parameter
         # if ($PSBoundParameters.ContainsKey('IdentityType')) {
-        if ($allPoliciesWithVault.count -gt 0) {
+        if ($allPoliciesWithVault.count -gt 0) {
```
[the remark this answers is not in the document]
Comment (TracyYaoYao) [reply src/Cdn/custom/Start-AzFrontDoorCdnProfilePrepareMigration.ps1:300]: Thanks for your patience. Fix in the next commit.

File: src/Cdn/custom/Enable-AzFrontDoorCdnProfileMigration.ps1 [hunk 3]
```diff
@@ -117,6 +123,5 @@ function Enable-AzFrontDoorCdnProfileMigration {
         Write-Host("Start to migrate.")
         Write-Host("This process will disable your Front Door (classic) profile and move all your traffic and configurations to the new Front Door profile.")
         Az.Cdn.internal\Invoke-AzCdnCommitProfileMigration @PSBoundParameters
```
[the remark this answers is not in the document]
Comment (TracyYaoYao) [reply src/Cdn/custom/Enable-AzFrontDoorCdnProfileMigration.ps1:125]: I am not sure whether I got your words. I check the customized command which supports $PassThru, but no special processing of output result. Like: https://github.com/Azure/azure-powershell/blob/bfa82ee9df73f071ddcdea6baccf0f0eeb05d0a5/src/Cdn/custom/Remove-AzCdnProfile.ps1#LL135C1-L135C21

File: src/Cdn/custom/Remove-AzFrontDoorCdnProfile.ps1 [hunk 4]
```diff
@@ -155,21 +155,36 @@ function Remove-AzFrontDoorCdnProfile {
     )
 
     process {
+        $hasAsJob = $PSBoundParameters.Remove('AsJob')
+        $hasNoWait = $PSBoundParameters.Remove('NoWait')
+        $hasPassThru = $PSBoundParameters.Remove('PassThru')
```
Comment (msJinLei) [inline src/Cdn/custom/Remove-AzFrontDoorCdnProfile.ps1:160]: Specify `PassThru` in Remove-AzFrontDoorCdnProfile.Tests.ps1

File: src/Cdn/custom/Stop-AzFrontDoorCdnProfileMigration.ps1 [hunk 5]
```diff
@@ -123,7 +123,6 @@ function Stop-AzFrontDoorCdnProfileMigration {
         else {
             Remove-AzFrontDoorCdnProfile -ResourceGroupName ${ResourceGroupName} -Name ${ProfileName}
         }
-
-        Write-Host("Abort succeeded.")
+        Write-Host("Abort migration succeeded.")
```
Comment (msJinLei) [inline src/Cdn/custom/Stop-AzFrontDoorCdnProfileMigration.ps1:126]:
```suggestion
        Write-Host("Migration abort succeeded.")
```

File: src/Cdn/custom/Update-AzFrontDoorCdnProfile.ps1 [hunk 6]
```diff
@@ -179,37 +179,51 @@ function Update-AzFrontDoorCdnProfile {
     process {
         $hasTag = $PSBoundParameters.Remove('Tag')
         $hasOriginResponseTimeout = $PSBoundParameters.Remove('OriginResponseTimeoutSecond')
+        $hasIdentityType = $PSBoundParameters.Remove('IdentityType')
+        $hasIdentityUserAssignedIdentity = $PSBoundParameters.Remove('IdentityUserAssignedIdentity')
```
Comment (msJinLei) [inline src/Cdn/custom/Update-AzFrontDoorCdnProfile.ps1:183]: Please specify `IdentityUserAssignedIdentity` and `OriginResponseTimeoutSecond` in Update-AzFrontDoorCdnProfile.Tests.ps1

File: src/Cdn/test/Remove-AzFrontDoorCdnProfile.Tests.ps1 [hunk 7]
```diff
@@ -24,7 +24,22 @@ Describe 'Remove-AzFrontDoorCdnProfile'  {
...
+    It 'Delete' {
+        $frontDoorCdnProfileName = 'fdp-pstest020'
+        Write-Host -ForegroundColor Green "Use frontDoorCdnProfileName : $($frontDoorCdnProfileName)"
+
+        $profileSku = "Standard_AzureFrontDoor"
+        Write-Host -ForegroundColor Green "New frontDoorCdnProfileName"
+        New-AzFrontDoorCdnProfile -SkuName $profileSku -Name $frontDoorCdnProfileName -ResourceGroupName $env.ResourceGroupName -Location Global
+
+        Write-Host -ForegroundColor Green "Remove frontDoorCdnProfileName"
+        $res = Remove-AzFrontDoorCdnProfile -Name $frontDoorCdnProfileName -ResourceGroupName $env.ResourceGroupName -PassThru
```
Comment (msJinLei) [inline src/Cdn/test/Remove-AzFrontDoorCdnProfile.Tests.ps1:41]: `$PassThru` added

File: tools/StaticAnalysis/Exceptions/Az.Cdn/BreakingChangeIssues.csv [hunk 8]
```diff
@@ -0,0 +1,2 @@
+"Module","ClassName","Target","Severity","ProblemId","Description","Remediation"
```
Comment (msJinLei) [inline tools/StaticAnalysis/Exceptions/Az.Cdn/BreakingChangeIssues.csv:1]: > [Microsoft.Azure.PowerShell.Cmdlets.Cdn.Runtime.PreviewMessageAttribute("This cmdlet is using a preview API version and is subject to breaking change in a future release.")]

Cmdlet is in preview

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by msJinLei** [comment]
@TracyYaoYao Please provide changelog as a comment here for you PR. We include it into main branch.

File: src/Cdn/custom/Remove-AzCdnProfile.ps1 [hunk 9]
```diff
@@ -155,21 +155,37 @@ function Remove-AzCdnProfile {
...
+        $hasNoWait = $PSBoundParameters.Remove('NoWait')
+        $hasPassThru = $PSBoundParameters.Remove('PassThru')
```
[the remark this answers is not in the document]
Comment (TracyYaoYao) [reply src/Cdn/custom/Remove-AzCdnProfile.ps1:160]: `PassThru` has added as a single testcase in the Remove-AzCdnProfile.Tests.ps1 
https://github.com/TracyYaoYao/azure-powershell/blob/2a213ccb06040ef7f38fbf1d264899831c018fba/src/Cdn/test/Remove-AzCdnProfile.Tests.ps1#LL32C7-L32C7

File: src/Cdn/custom/Remove-AzFrontDoorCdnProfile.ps1 (hunk 4 above)
Comment (TracyYaoYao) [reply src/Cdn/custom/Remove-AzFrontDoorCdnProfile.ps1:160]: Same to Remove-AzFrontDoorCdnProfile.Tests.ps1, `PassThru` has added as a single testcase 
https://github.com/TracyYaoYao/azure-powershell/blob/2a213ccb06040ef7f38fbf1d264899831c018fba/src/Cdn/test/Remove-AzFrontDoorCdnProfile.Tests.ps1#LL32C7-L32C7

**Comment by TracyYaoYao** [comment]
Az.Cdn 3.0.1
1. Upgrade API version to 2023-05-01
2. Fix known issue for `Update-AzCdnProfile`, `Update-AzFrontDoorCdnProfile`, `Remove-AzCdnProfile`, `Remove-AzCdnProfile`

File: src/Cdn/custom/Update-AzFrontDoorCdnProfile.ps1 (hunk 6 above)
Comment (TracyYaoYao) [reply src/Cdn/custom/Update-AzFrontDoorCdnProfile.ps1:183]: `OriginResponseTimeoutSecond` added, `IdentityUserAssignedIdentity ` will be added into Live test in main branch.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**